source/io/HttpClient.cpp

```diff
@@ -76,6 +78,11 @@ void GenerateNotSupportedOnPlatformError(ErrorCluster *errorCluster, ConstCStr m
 VIREO_FUNCTION_SIGNATURE6(HttpClientOpen, StringRef, StringRef, StringRef, UInt32, UInt32, ErrorCluster)
 {
 #if kVireoOS_emscripten
+    if (!_ParamPointer(0) || !_ParamPointer(1) || !_ParamPointer(2) || !_ParamPointer(3) || !_ParamPointer(4) || !_ParamPointer(5)) {
```

Comment:
Agreed the checks are too aggressive and should be fixed in the future to better align with the optional / required pattern

As this implementation improves on previous behavior (where no null checking is performed and memory could be corrupt), and does not impact tests, or current uses I am leaving TODOs and creating an issue to track the work